### ISSUE TITLE
Unhandled rejection can expose GC to user code

### ISSUE THREAD

**Issue by mhofman** [Agoric/Agoric#8191]
## Describe the bug
The unhandled rejection logic in xsnap-worker is to [invoke `console.error` in the start compartment](https://github.com/agoric-labs/xsnap-pub/blob/87566073e377cbc6fe3f874e6cbed990de3f07b6/xsnap/sources/xsnapPlatform.c#L168-L177) (supervisor), when the unhandled promise is GCed. Because of the workaround for #6784, and the fact XS internally [keeps these unhandled promises in a WeakRef](https://github.com/agoric-labs/moddable/blob/f6c5951fc055e4ca592b9166b9ae3cbb9cca6bf0/xs/sources/xsPromise.c#L214-L219), these are only collected during BOYD.

So far, so good, all is deterministic, but unfortunately `console.error` uses `object-inpect.js` to issue a command to log the unhandled rejection, which may invoke user code if the reason value of the rejection can trap (e.g. using accessors or a proxy), which means the user code is now capable of sensing gc. It also means user code is potentially executing during the BOYD delivery.

## Expected behavior
User code is not able to sense gc through unhandled rejections

**Comment by mhofman** [comment]
One solution may be to harden `object-inspect.js` to not trigger any user code. That would require jumping through hoops to only traverse non-proxy objects and only data properties.

**Comment by gibson042** [comment]
I think that would require many changes to object-inspect.js and/or replacing the native `Proxy` in workers, but it does feel like the right approach.

**Comment by mhofman** [comment]
Yes, we are already planning on replacing the `Proxy` constructor available to vat code with a stamping one, and gain an `IsProxy` predicate (the original motivations is `passStyleOf` checks). See https://github.com/Agoric/agoric-sdk/issues/3905
